desc/compute/data_index.py

```diff
@@ -75,7 +80,55 @@ def _decorator(func):
             "coordinates": coordinates,
             "dependencies": deps,
         }
-        data_index[name] = d
+        for p in parameterization:
+            flag = False
+            for base_class, superclasses in _class_inheritance.items():
+                if p in superclasses or p == base_class:
+                    data_index[base_class][name] = d.copy()
+                    flag = True
+            if not flag:
+                raise ValueError(
+                    f"Can't register function with unknown parameterization: {p}"
+                )
         return func
 
     return _decorator
+
+
+# This allows us to handle subclasses whos data_index stuff should inherit
+# from parent classes.
+# This is the least bad solution I've found, since everything else requires
+# crazy circular imports
+# could maybe make this fancier with a registry of compute-able objects?
+_class_inheritance = {
+    "desc.equilibrium.equilibrium.Equilibrium": [],
```

Comment:
Wouldn't it be more natural to have the keys of this dictionary be the superclass and the values be the subclasses? Then if the parameteriztaion is a superclass (key in this dict) then assign `d.copy()` to all values (subclasses). If the parameteriztaion is a subclass then just assign `d.copy()` to that subclass.